contrib/common.sh

```diff
@@ -0,0 +1,80 @@
+# Common functions for wrapper scripts for xdg-desktop-portal-termfilechooser.
+
+# Input args and ouptut format of the wrapper scripts:
+#
+# Inputs:
+# 1. "1" if multiple files can be chosen, "0" otherwise.
+# 2. "1" if a directory should be chosen, "0" otherwise.
+# 3. "0" if opening files was requested, "1" if writing to a file was
+#    requested. For example, when uploading files in Firefox, this will be "0".
+#    When saving a web page in Firefox, this will be "1".
+# 4. If writing to a file, this is recommended path provided by the caller. For
+#    example, when saving a web page in Firefox, this will be the recommended
+#    path Firefox provided, such as "~/Downloads/webpage_title.html".
+#    Note that if the path already exists, we keep appending "_" to it until we
+#    get a path that does not exist.
+# 5. The output path, to which results should be written.
+#
+# Output:
+# The script should print the selected paths to the output path (argument #5),
+# one path per line.
+# If nothing is printed, then the operation is assumed to have been canceled.
+
+
+default_termcmd () {
+	# Picking the default terminal emulator. Similar to
+	# 	https://github.com/i3/i3/blob/next/i3-sensible-terminal
+	# The first available command is used.
```

Comment:
This is not greate if user have more than 1 terminal. For example: I have both wezterm and kitty. But I only want to use kitty for with portal. So it better for user to set an `TERMCMD`.

Reply:
Nothing stops a user from setting their preferred terminal emulator manually with `$termcmd`. But in case the user did not choose explicitly, this function picks a term that is actually installed.
Because what is the alternative? Hard-coding some value for everyone?

Reply:
No, it doesn't. `termcmd=$(default_termcmd)`. If `default_termcmd` doesn't find any terminal, nothing is echoed, and termcmd is set to empty.
![image](https://github.com/user-attachments/assets/66dc4d4f-60a6-4f5c-ba11-c39ba5a44d52)

Reply:
The `termcmd=…` line was meant to be edited in the wrapper script, not set on the command line. To use the global environment value, the line can be changed to

	termcmd=${termcmd:-$(default_termcmd)}

This way if the `$termcmd` variable is set with

	❯ export termcmd=<…>

its value `<…>` will be used; otherwise it will be obtained from the `default_termcmd()` function.